include/aws/common/math.inl

```diff
@@ -53,6 +53,13 @@ AWS_EXTERN_C_BEGIN
 #    pragma warning(disable : 4127) /*Disable "conditional expression is constant" */
 #endif                              /* _MSC_VER */
 
+#ifdef __cplusplus
+#    if defined(__GNUC__)
+#        pragma GCC diagnostic push
```

Comment:
You also need to `pop`
Otherwise, any code that includes the .h file that includes this .inl file will have these warnings disabled